src/coreComponents/constitutive/relativePermeability/TableRelativePermeabilityHysteresis.hpp

```diff
@@ -359,42 +369,155 @@ class TableRelativePermeabilityHysteresis : public RelativePermeabilityBase
+    static constexpr char const * imbibitionPhaseMinVolumeFractionString()
+    { return "imbibitionPhaseMinVolumeFraction"; }
 
-    static constexpr char const * drainageWettingNonWettingRelPermTableNamesString() { return "drainageWettingNonWettingRelPermTableNames"; }
-    static constexpr char const * drainageWettingIntermediateRelPermTableNamesString() { return "drainageWettingIntermediateRelPermTableNames"; }
-    static constexpr char const * drainageNonWettingIntermediateRelPermTableNamesString() { return "drainageNonWettingIntermediateRelPermTableNames"; }
+    static constexpr char const * drainagePhaseMaxVolumeFractionString()
+    { return "drainagePhaseMaxVolumeFraction"; }
 
-    static constexpr char const * imbibitionWettingRelPermTableNameString() { return "imbibitionWettingRelPermTableName"; }
-    static constexpr char const * imbibitionNonWettingRelPermTableNameString() { return "imbibitionNonWettingRelPermTableName"; }
+    static constexpr char const * imbibitionPhaseMaxVolumeFractionString()
+    { return "imbibitionPhaseMaxVolumeFraction"; }
+
+    static constexpr char const * drainageWettingNonWettingRelPermTableNamesString()
+    { return "drainageWettingNonWettingRelPermTableNames"; }
+
+    static constexpr char const * drainageWettingIntermediateRelPermTableNamesString()
+    { return "drainageWettingIntermediateRelPermTableNames"; }
+
+    static constexpr char const * drainageNonWettingIntermediateRelPermTableNamesString()
+    { return "drainageNonWettingIntermediateRelPermTableNames"; }
+
+    static constexpr char const * imbibitionWettingRelPermTableNameString()
+    { return "imbibitionWettingRelPermTableName"; }
+
+    static constexpr char const * imbibitionNonWettingRelPermTableNameString()
+    { return "imbibitionNonWettingRelPermTableName"; }
 
   };
 
+  arrayView1d< real64 const > getPhaseMinVolumeFraction() const override
+  { return m_imbibitionPhaseMinVolFraction; };
+
+
+  /// for use in RelpermDriver to browse the drainage curves
+  /// by setting the MaxHistoricalNonWettingSat to Snwmin and MinWettingSat to Sw
+  GEOSX_HOST_DEVICE
+  std::tuple< integer, integer, integer > getPhaseIndex() const
+  {
+    using PT = PhaseType;
+    integer const ipWater = m_phaseOrder[PT::WATER];
+    integer const ipOil = m_phaseOrder[PT::OIL];
+    integer const ipGas = m_phaseOrder[PT::GAS];
+
+    integer ipWetting = -1, ipNonWetting = -1, ipInter = -1;
+
+    if( ipWater >= 0 && ipOil >= 0 && ipGas >= 0 )
+    {
+      ipWetting = ipWater;
+      ipInter = ipOil;
+      ipNonWetting = ipGas;
+    }
+    else if( ipWater < 0 )
+    {
+      ipWetting = ipOil;
+      ipNonWetting = ipGas;
+    }
+    else if( ipOil < 0 )
+    {
+      ipWetting = ipWater;
+      ipNonWetting = ipGas;
+    }
+    else if( ipGas < 0 )
+    {
+      ipWetting = ipWater;
+      ipNonWetting = ipOil;
+    }
+
+    //maybe a bit too pythonic
+    return std::make_tuple( ipWetting, ipInter, ipNonWetting );
+  }
```

Comment:
I would put this either in the `RelativePermeabilityBase` class so it can be by all the relperm classes, or directly in the `RelpermDriver` to avoid cluttering the already large `TableRelativePermeabilityHysteresis` class. 

Also, since this function is actually used to only get `ipWetting` and `ipNonWetting`, maybe there is no need to return `ipInter`, especially when it is -1.

Reply:
Done